src/mapping-pmp-v0.ts

```diff
@@ -0,0 +1,250 @@
+      pmpInput.key,
+      latestNonce
+    );
+
+    let pmp = new PMP(pmpId);
+    pmp.key = pmpInput.key;
+    if (token) {
+      // @dev it is valid for PMP to be created before token exists
+      pmp.token = token.id;
+    }
+
+    let currentConfiguredParamType = getPMPParamTypeString(
+      BigInt.fromI32(pmpInput.configuredParamType)
+    );
+
+    pmp.tokenPMPNonce = latestNonce;
+    pmp.configuredParamType = currentConfiguredParamType;
+    pmp.configuredValue = pmpInput.configuredValue.toHexString();
+
+    if (pmpInput.configuringArtistString) {
+      pmp.artistConfiguredValueString = pmpInput.configuredValueString;
+    } else {
+      pmp.nonArtistConfiguredValueString = pmpInput.configuredValueString;
+    }
+
+    // update the token pmp
+    pmp.createdAt = event.block.timestamp;
+    pmp.save();
+
+    // update the latest state entity
+    latestState.latestNonce = latestNonce;
+    latestState.save();
+  }
+}
+
+export function handleProjectConfigured(event: ProjectConfigured): void {
+  let projectId = generateContractSpecificId(
+    event.params.coreContract,
+    event.params.projectId
+  );
+  let project = Project.load(projectId);
+
+  if (!project) {
+    // @dev While PMP is open to any NFT, we only index params on indexed projects
+    return;
+  }
+  const projectConfigId = generateProjectConfigId(
+    event.address,
+    event.params.coreContract,
+    event.params.projectId
+  );
+
+  let projectConfig = PMPProjectConfig.load(projectConfigId);
+  if (!projectConfig) {
+    projectConfig = new PMPProjectConfig(projectConfigId);
+    projectConfig.project = project.id;
+
+    // add project pmp config to the project
+    project.pmpProjectConfig = projectConfig.id;
+    project.save();
+  } else {
+    // clean up old pmp config, only store the most recent configurations
+    let previousPMPConfigCount = projectConfig.pmpConfigCount;
+    let storedKeys = projectConfig.pmpConfigKeys;
+
+    if (previousPMPConfigCount.toI32() != storedKeys.length) {
+      log.warning(
+        "Inconsistent data found for project config {}: count is {} but keys array length is {}",
+        [
+          projectConfigId,
+          previousPMPConfigCount.toString(),
+          storedKeys.length.toString()
+        ]
+      );
+    } else if (previousPMPConfigCount.gt(BigInt.fromI32(0))) {
+      // delete all old pmp configs
+      for (let i = 0; i < previousPMPConfigCount.toI32(); i++) {
+        const prevPMPConfigId = storedKeys[i];
+        store.remove("PMPConfig", prevPMPConfigId);
+      }
+    }
+  }
+
+  // create PMPConfig entities for each input config
+  let configKeys: string[] = [];
+  for (let i = 0; i < event.params.pmpInputConfigs.length; i++) {
+    const config = event.params.pmpInputConfigs[i];
+    const pmpConfigId = generatePMPConfigId(
+      event.address,
+      event.params.coreContract,
+      event.params.projectId,
+      config.key
+    );
+
+    configKeys.push(pmpConfigId);
```

Comment:
For consistency with the on-chain structs, I was expecting this would just store the PMP key value, and we would build the `prevPMPConfigId` when deleting, but this works just the same. if we do it this way, maybe it should be called `pmpConfigIds` in the schema?

Reply:
ah makes sense! I had it that way at first and then updated it to the full id to reduce calls to the id builder helper function but prefer consistency with on-chain structs so reverting that back to just the key value!